frontend/src/components/templates/Form/index.tsx

```diff
@@ -191,6 +194,21 @@ export const Form: React.FC<FormProps> = ({
     }
   }, [successMessages, errorMessages, alertRef]);
 
+  // Display a warning popup if the demandeur does not have a phone number 📞
+  useEffect(() => {
+    if (!isEmpty(enrollment.team_members)) {
+      const firstDemandeur = enrollment.team_members?.find(
+        ({ type }) => type === 'demandeur'
+      );
+
+      if (firstDemandeur?.phone_number) {
+        setDemandeurPhoneNumberWarning(false);
+      } else {
+        setDemandeurPhoneNumberWarning(true);
+      }
+    }
+  }, [enrollment]);
```

Comment:
I am not sure to understand why you use a `useEffect` here rather than just compute  `demandeurPhoneNumberWarning` on render

I feel like it aint gonna change but maybe I dont get the data cycle here

Reply:
Yes it can change as the user is pleased to update his `phone_number` on Mon Compte Pro !